account_mtd/models/mtd_hello_world.py

```diff
@@ -0,0 +1,406 @@
+                '/web#id={}&view_type=form&model=mtd.hello_world'.format(self.id) +
+                '&menu_id={}&action={}'.format(menu_id.id, action.id))
+            
+    @api.multi        
+    def exchange_user_authorisation(self, auth_code, record_id, tracker_id):
+        self._logger.info("(Step 2) exchange authorisation code")
+        api_tracker = self.env['mtd.api_request_tracker'].search([('id', '=', tracker_id)])
+        api_token = self.env['mtd.api_tokens'].search([('api_id', '=', api_tracker.api_id)])
+
+        if api_token:
+            api_token.authorisation_code = auth_code
+        else:
+            api_token.create({
+                'api_id': api_tracker.api_id,
+                'api_name': api_tracker.api_name,
+                'authorisation_code': auth_code
+                })
+        # self.current_record = record_id
+        record = self.env['mtd.hello_world'].search([('id', '=', record_id)])
+        token_location_uri = "https://test-api.service.hmrc.gov.uk/oauth/token"
+        client_id = record.hmrc_configuration.client_id
+        client_secret = record.hmrc_configuration.client_secret
+        redirect_uri = "http://localhost:8090{}".format('/auth-redirect')
+
+        data_user_info = {
+            'grant_type': 'authorization_code',
+            'client_id': client_id,
+            'client_secret': client_secret,
+            'redirect_uri': redirect_uri,
+            'code': auth_code
+            }
+        self._logger.info(
+            "(Step 2) exchange authorisation code - Data which will be " +
+            "sent in the request:-  {}".format(json.dumps(data_user_info))
+        )
+        headers = {
+            'Content-Type': 'application/json',
+            }
+        self._logger.info(
+            "(Step 2) exchange authorisation code - headers which will be "
+            "sent in the request:-  {}".format(headers)
+        )
+        req = requests.post(token_location_uri, data=json.dumps(data_user_info), headers=headers)
+        response_token = json.loads(req.text)
+        self._logger.info(
+            "(Step 2) exchange authorisation code - received " 
+            "response of the request:- {}, and its text:- {}".format(req, response_token)
+        )
+        if req.ok:
+            # get the record which we created when sending the request and update the response_received column
+            # As this determines whether we can place another request or not
+            record_tracker = self.env['mtd.api_request_tracker'].search([
+                ('request_sent', '=', True),
+                ('response_received', '=', False)
+            ])
+            record_tracker.response_received = True
+            if not api_token:
+                api_token = self.env['mtd.api_tokens'].search([('authorisation_code', '=', auth_code)])
+            self._logger.info(
+                "(Step 2) exchange authorisation code - api_token table id where info is stored:- {}".format(api_token)
```

Comment:
I suggest sticking the `.format(api_token)` on the next line, with the `.` in line with the opening `"`.  If the string above is more than 80 characters after that, split it before the ` - `. 

With both tweaks it would look something like this:

```
self._logger.info(
    "(Step 2) ... code"
    " - api_token table id ...:-{}"
    .format(api_token)
)
```

Once can read straight down and see that we're doing a string formatting operation.
Remember that 80 characters is luxury considering the human visual cortex tends to prefer around 60 characters!